### DISCUSSION TITLE
Can't communicate SX1262 with Lorawan gateway's FSK channel

### DISCUSSION THREAD

**Discussion by tauesdospi** [jgromes/RadioLib#832] [General]
Hi,

I'm using heltec wifi stick lite which has an SX1276 modem. I'm using FSK for communication and it is working correctly:

SX1276 radio = new Module(GPIO_NUM_18, GPIO_NUM_26, GPIO_NUM_14, GPIO_NUM_35);
  state = radio.beginFSK();  
  state = radio.setFrequency(868.8);
  state = radio.setBitRate(50.0);
  state = radio.setFrequencyDeviation(25);
  state = radio.setRxBandwidth(50.0); 
  state = radio.setDataShaping(RADIOLIB_SHAPING_1_0);
  uint8_t syncWord[] = {0xC1, 0x94, 0xC1 };
  state = radio.setSyncWord(syncWord, 3);
  state = radio.setPreambleLength(40);
  state = radio.disableAddressFiltering();
  state = radio.setEncoding(RADIOLIB_ENCODING_WHITENING);
  radio.setDio0Action(OnFSKCompleted, RISING);
  state = radio.setAFCBandwidth(83.3);
  state = radio.setCRC(true);
  state = radio.variablePacketLengthMode();

With these settings I can communicate with a Lorawan gateway using the FSK channel defined in the EU868 specs.

Now I'm using the Wireless Mini Shell module also from Heltec (HT-CT62) which has an SX1262 modem and I'm unable to communicate.

I had to remove the setAFCBandwidth(x) as SX1262 doesn't have this method. Also setDio0Action is  setDio1Action on SX1262. I'm unable to start the SPI bus; it seems the advertised pins are not correct so I'm using the "Mcu.begin()" proprietary heltec method which configures the SPI (among other things).

The code is the same of above but replacing SX1276 with
SX1262 radio = new Module(RADIO_NSS, RADIO_DIO_1, RADIO_RESET, RADIO_BUSY);
(the pins are the same in "Lorawan_APP.h" which initializes the heltec lorawan library (which I'm not using). The pins are correct.

I have checked the SX1262 ping pong example and using the Mcu.begin() and the pins above, it works so the radio can tx/rx in FSK mode but for any reason I don't know why I can no longer communicate with the FSK channel of the lorawan gateway with the above settings (same settings for SX1276).

I'm asking for help at this point as I don't know how to fix this...
Thank you!
Antonio

**Comment by jgromes** [comment]
The first thing I would do is to check the return codes from all of the methods, you may find some of them have actually failed. You wouldn't see that in your test because if something is failing, it will likely fail the same way on the other SX126x node.

Next, the debug mode is also quite handy.

**Reply by tauesdospi** [reply]
Hi
thank you for your feedback!

 state = radio.setRxBandwidth(50.0); => this in SX1276 works correctly but in SX216x it is quantized (as per source code) and I have instead put 58.6

all functions setXXX return 0 ... so everything is correct but it doesn't work... it seems there is any other difference between SX1276 / SX1262 models ...

I have enabled both debug and verbose mode but for me it has not sense....

Any idea which other parameter may be required in SX1262 ? May be the parameter order? 

Regards
Antonio

**Reply by jgromes** [reply]
I think there is a different whitening between SX127x and SX126x. Try calling `setWhitening(true, 0x01FF);` instead of or after `setEncoding`.

**Reply by tauesdospi** [reply]
SI SEÑOR !  You are the master! Thank you very much for your continued help!
